test/libsolidity/InlineAssembly.cpp

```diff
@@ -170,7 +170,7 @@ BOOST_AUTO_TEST_CASE(smoke_test)
 
 BOOST_AUTO_TEST_CASE(surplus_input)
 {
-	CHECK_PARSE_ERROR("{ } { }", ParserError, "Expected token EOS");
+	CHECK_PARSE_ERROR("{ } { }", ParserError, "Expected end of source but got '{'");
```

Comment:
We could think about moving these to inline assembly blocks in actual contracts in regular parsing tests (e.g. tests in syntaxTests/parsing) instead.

Reply:
I think it may not be possible yet to test the strict assembly, I have a change to support that in the language, then we can move on it.

Reply:
This is something that can only be done for assembly-only mode, since only there we know where to expect end-of-source. In inline assembly mode, the parsing continues in solidity mode. Example:

```
assembly {} {}
```

So this would be valid source (the second `{}` is just an empty solidity block).

Perhaps I misunderstood your comment.